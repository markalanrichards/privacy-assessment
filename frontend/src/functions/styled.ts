import * as styledComponents from "styled-components";

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider
} = styledComponents as styledComponents.ThemedStyledComponentsModule<
  IThemeInterface
>;

interface Color {
  primaryColor: string;
}

interface Size {
  large: string;
}

export interface IThemeInterface {
  color: Color;
  size: Size;
}

export const defaultTheme: IThemeInterface = {
  color: {
    primaryColor: "yellow"
  },
  size: {
    large: "10"
  }
};

export default styled;
export { css, injectGlobal, keyframes, ThemeProvider };
