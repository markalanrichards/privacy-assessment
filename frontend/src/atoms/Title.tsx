import styled from "styled-components";

const Title = styled.h1`
  color: ${props => props.theme.color.color};
  font-size: ${props => props.theme.size.large};
`;

const Super = Title.extend`
  font-size: ${props => props.theme.size.xxl};
  text-transform: uppercase;
`;
export { Super, Title as Normal };
