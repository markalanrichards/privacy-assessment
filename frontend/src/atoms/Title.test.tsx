import * as React from "react";
import { shallow, mount } from "enzyme";
import styled, { ThemeProvider } from "styled-components";
import "jest-styled-components";

import { Normal, Super } from "./Title";

const defaultTheme = {
  color: {
    color: "yellow"
  },
  size: {
    large: "10"
  }
};

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    // @ts-ignore
    .getChildContext(); // not in enzyme types file GRRRR!!
  return shallow(tree, { context });
};

describe("Title component With Theme", () => {
  test("Normal", () => {
    const wrapper = shallowWithTheme(<Normal>title</Normal>, defaultTheme);
    // throw wrapper.debug()
    expect(wrapper).toMatchSnapshot();
  });

  test("Super", () => {
    const wrapper = shallowWithTheme(<Super>title</Super>, defaultTheme);
    // throw wrapper.debug()
    expect(wrapper).toMatchSnapshot();
  });
});
