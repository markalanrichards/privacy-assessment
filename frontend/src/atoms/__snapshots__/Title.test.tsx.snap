// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component With Theme Normal 1`] = `
.c0 {
  color: yellow;
  font-size: 10;
}

<h1
  className="c0"
>
  title
</h1>
`;

exports[`Title component With Theme Super 1`] = `
.c0 {
  color: yellow;
  font-size: 10;
  text-transform: uppercase;
}

<h1
  className="c0"
>
  title
</h1>
`;
